### DISCUSSION TITLE
incomplete type `uct_device_addr_t`?

### DISCUSSION THREAD

**Discussion by Honggang-LI** [openucx/ucx#7561] [Q&A]
https://github.com/openucx/ucx/blob/59998e0fd8e3620b222b8b4fca4fcf078512c3b7/src/uct/api/uct_def.h#L96

I can't find the definition of `struct uct_device_addr`. 

```
typedef struct uct_device_addr     uct_device_addr_t;
typedef struct uct_iface_addr      uct_iface_addr_t;
typedef struct uct_ep_addr         uct_ep_addr_t;
```

It seems those three struct are incomplete types. Why?

thanks

**Comment by Honggang-LI** [comment]


**Reply by yosefe** [reply]
the structs are just placeholders and the actual struct of device/iface/ep address is defined by each transport

**Reply by Honggang-LI** [reply]
> the structs are just placeholders and the actual struct of device/iface/ep address is defined by each transport

thanks
